lib/Drush/Commands/core/SiteInstallCommands.php

```diff
@@ -0,0 +1,283 @@
+   * @param string $profile The install profile you wish to run. Defaults to 'default' in D6, 'standard' in D7+, unless an install profile is marked as exclusive (or as a distribution in D8+ terminology) in which case that is used.
+   * @param string $additional Any additional settings you wish to pass to the profile. The key is in the form [form name].[parameter name]
+   * @option db-url A Drupal 6 style database URL. Only required for initial install - not re-install.
+   * @option db-prefix An optional table prefix to use for initial install.  Can be a key-value array of tables/prefixes in a drushrc file (not the command line).
+   * @option db-su Account to use when creating a new database. Must have Grant permission (mysql only). Optional.
+   * @option db-su-pw Password for the "db-su" account. Optional.
+   * @option account-name uid1 name. Defaults to admin
+   * @option account-pass uid1 pass. Defaults to a randomly generated password. If desired, set a fixed password in drushrc.php.
+   * @option account-mail uid1 email. Defaults to admin@example.com
+   * @option locale A short language code. Sets the default site language. Language files must already be present.
+   * @option site-name Defaults to Site-Install
+   * @option site-mail From: for system mailings. Defaults to admin@example.com
+   * @option sites-subdir Name of directory under 'sites' which should be created. Only needed when the subdirectory does not already exist. Defaults to 'default'
+   * @option config-dir A path pointing to a full set of configuration which should be imported after installation.
+   * @usage drush site-install expert --locale=uk
+   *   (Re)install using the expert install profile. Set default language to Ukrainian.
+   * @usage drush site-install --db-url=mysql://root:pass@localhost:port/dbname
+   *   Install using the specified DB params.
+   * @usage drush site-install --db-url=sqlite://sites/example.com/files/.ht.sqlite
+   *   Install using SQLite
+   * @usage drush site-install --account-name=joe --account-pass=mom
+   *   Re-install with specified uid1 credentials.
+   * @usage drush site-install standard install_configure_form.site_default_country=FR my_profile_form.my_settings.key=value
+   *   Pass additional arguments to the profile (D7 example shown here.
+   * @usage drush site-install standard install_configure_form.update_status_module='array(FALSE,FALSE)'
+   *   Disable email notification during install and later. If your server has no smtp, this gets rid of an error during install.
+   * @bootstrap DRUSH_BOOTSTRAP_DRUPAL_ROOT
+   * @aliases si
+   *
+   */
+  public function install($profile, array $additional = [], $options = ['db-url' => NULL, 'db-prefix' => NULL, 'db-su' => NULL, 'db-su-pw' => NULL, 'account-name' => 'admin', 'account-mail' => 'admin@example.com', 'account-pass' => NULL, 'locale' => 'en', 'site-name' => 'Drush Site-Install', 'site-pass' => NULL, 'sites-subdir' => NULL, 'config-dir' => NULL]) {
```

Comment:
This declaration for variable arguments (the last declared `array`) works in consolidation/annotated-command and Robo. If you are having trouble with it here, the problem is in the annotation adapter. I'll look at that when I get a chance.